src/Actions/OrderPosition/CreateOrderPosition.php

```diff
@@ -82,6 +83,39 @@ public function performAction(): OrderPosition
         unset($orderPosition->discounts);
         $orderPosition->save();
 
+        if ($this->data['product_id'] ?? false) {
+            $product = $orderPosition->product()->with('bundleProducts')->first();
+            $sortNumber = $orderPosition->sort_number;
+            $product->bundleProducts
+                ->map(function (Product $bundleProduct) use ($orderPosition, &$sortNumber) {
+                    $sortNumber++;
```

Comment:
this can screw the ordering, because there could be positions in this order with the same sort_number.
Therefore all other positions with higher sort_numbers than the initial order position should be updated accordingly.
Maybe you can find a satisfying solution for this issue.